src/libcore/cell.rs

```diff
@@ -1038,7 +1042,7 @@ impl<'b> Drop for BorrowRef<'b> {
     #[inline]
     fn drop(&mut self) {
         let borrow = self.borrow.get();
-        debug_assert!(borrow < MIN_WRITING && borrow != UNUSED);
+        debug_assert!(borrow > UNUSED);
         self.borrow.set(borrow - 1);
```

Comment:
Oh wow and this can overflow again to a positive value so you can even keep using the `RefCell`... crazy.

Reply:
Yep :)

Reply:
Maybe be worth a comment though. (If you decide to reinstate that, that is.)

Reply:
You mean if we decide to allow overflow? It sounds like the decision was not to allow it.

Reply:
Well I figured that'll depend on perf results.

Reply:
True. If we go back to allowing overflow, I'll add a comment here.